### DISCUSSION TITLE
What is the hook or process that runs just once whenever the tests rerun?

### DISCUSSION THREAD

**Discussion by flowt-au** [vitest-dev/vitest#6233] [Q&A]
I want to reseed my database via an API call before any test files run. I have tried all the various combinations of globalSetup, setupFiles, beforeAll(), beforeEach(), etc and cannot seem to get this working. 

I start my tests with the following script: `"test:unit": "vitest"` via `yarn test:unit`. The tests run and the watcher kicks in. The globalSetup file runs the first time I do that, but each time I press 'r', or change a file that triggers "RERUN", globalSetup does not run again.

As per the docs, the setupFile runs before each test file, and globalThis seems to be reset for each file so setting a flag to say "you have already reseeded the db" wont work.

So, what is the way to run some code before any test files run and whenever the RERUN is triggered? I must be missing something obvious here but cannot see what.

Thanks,
Murray

**Comment by junajan** [comment]
HI @flowt-au .. did you find a solution to this?

**Reply by flowt-au** [reply]
Not directly, no. I am now using [https://github.com/amiceli/vitest-cucumber](https://github.com/amiceli/vitest-cucumber) which is a great package. It supports a `BeforeAllScenarios` hook and in there I do all my setup.

```js
describeFeature(featureFile, ({ Background, BeforeAllScenarios, AfterAllScenarios, BeforeEachScenario, AfterEachScenario, ScenarioOutline, Scenario }) => {

  BeforeAllScenarios(async () => {
    // Do my db setup, async.
    await setupServerForTests(basetestdata, useServerForTest)
  })

  ScenarioOutline(`As an Admin User I want to add a new person to an Organisation`, ({ Given, When, Then, And }, variables) => {
     // etc ...
  })

})
